### DISCUSSION TITLE
understanding-your-middleware-pipeline-in-dotnet-6-with-the-middleware-analysis-package/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#138] [Announcements]
# understanding-your-middleware-pipeline-in-dotnet-6-with-the-middleware-analysis-package/

In this post I introduce the Microsoft.AspNetCore.MiddlewareAnalysis package and show how to use it to visualise the middleware in your .NET 6 apps.

https://andrewlock.net/understanding-your-middleware-pipeline-in-dotnet-6-with-the-middleware-analysis-package/

**Comment by harrisonogle** [comment]
Thanks for the article 👍.

In more recent versions of Microsoft.AspNetCore.MiddlewareAnalysis, the anonymous objects used when the diagnostic events are fired look take this structure ([source](https://github.com/dotnet/aspnetcore/blob/bec278eabea54f63da15e10e654bdfa4168a2479/src/Middleware/MiddlewareAnalysis/src/AnalysisMiddleware.cs#L53-L56)):
```C#
            _diagnostics.Write(
                "Microsoft.AspNetCore.MiddlewareAnalysis.MiddlewareStarting",
                new
                {
                    name = _middlewareName,
                    httpContext = httpContext,
                    instanceId = _instanceId,
                    timestamp = startTimestamp,
                });
```

In that case, the method signatures for `OnMiddlewareStarting`, `OnMiddlewareException`, and `OnMiddlewareFinished` in `AnalysisDiagnosticAdapter` should include a `Guid instanceId` parameter (as opposed to `Guid instance`), due to the way DiagnosticAdapter handles the anonymous objects.

Love the articles 👍👍

**Comment by Coelho04** [comment]
It seens that when using the minimal api the middlewareanalysis package it's not sending any events for your custom middleware and I believe the reason is because the middlewareanalysis is looking for a configure method that's currently does not exists.

I haven't tried but one possible solution is to register every custom middleware like in this example https://andrewlock.net/using-pathbase-with-dotnet-6-webapplicationbuilder/.

I'm missing something or doing something?
Is there any other way?
